assets/data-port/import/data/selectors.js

```diff
@@ -0,0 +1,91 @@
+import { steps } from '../steps';
+import { findKey } from 'lodash';
+
+/**
+ * Is fetching importer data selector.
+ *
+ * @param {Object} state Current state.
+ *
+ * @return {boolean} Is fetching.
+ */
+export const isFetching = ( state ) => state.isFetching;
+
+/**
+ * Fetch importer error selector.
+ *
+ * @param {Object} state Current state.
+ *
+ * @return {Object|boolean} Error object or false.
+ */
+export const getFetchError = ( state ) => state.fetchError;
+
+/**
+ * Step state selector.
+ *
+ * @param {Object}  state         Current state.
+ * @param {string}  step          Step name.
+ * @param {boolean} shouldResolve Flag whether should invoke the resolver.
+ *
+ * @return {Object} Step data.
+ */
+/* eslint-enable */
+export const getStepData = ( state, step ) => state[ step ];
+
+/**
+ * Get navigation steps with their state.
+ *
+ * @param {Object} state current state.
+ *
+ * @return {Array} Navigation steps.
+ */
+export const getNavigationSteps = ( { completedSteps } ) => {
+	const navSteps = steps.map( ( step ) => ( {
+		...step,
+		isComplete: completedSteps.includes( step.key ),
+		isNext: false,
+	} ) );
+
+	const nextStep =
+		navSteps.find( ( step ) => ! step.isComplete ) || navSteps[ 0 ];
+	nextStep.isNext = true;
+
+	return navSteps;
+};
+
+/**
+ * Get whether step is complete or not.
+ *
+ * @param {Object} state Current state.
+ * @param {string} step  Step name.
+ *
+ * @return {boolean} Step complete.
+ */
+export const isCompleteStep = ( { completedSteps }, step ) =>
+	completedSteps.includes( step );
+
+/**
+ * Get whether the importer is ready to start.
+ *
+ * @param {Object} state Current state.
+ *
+ * @return {boolean} If the importer is ready.
+ */
+export const isReadyToStart = ( state ) => {
+	const firstUploadedKey = findKey( state.upload, function( level ) {
```

Comment:
This could be simplified a bit as a search with `some` should be enough. And using the levels list from `levels.js` makes sure we are looking in the right objects:
 ```
levels
  .map( ( { key } ) => state.upload[key] )
  .some( ( level ) => level.isUploaded && ! level.inProgress  )
```

Reply:
That is a thing of beauty. I like it so much more. 20c4196d25b8c90c2ebefc0915e33ff86d138a6c